File: C#/square.cs
using System;

namespace SquareProg
{
    public class Square
    {
        private double x1, y1;
        private double x2, y2;
        private double x3, y3;
        private double x4, y4;

        public Square()
        {
            x1 = y1 = x2 = y2 = x3 = y3 = x4 = y4 = 0.0;
        }
        public Square(double x1, double y1, double x2, double y2, double x3, double y3, double x4, double y4)
        {
            this.x1 = x1;
            this.x2 = x2;
            this.x3 = x3;
            this.x4 = x4;
            this.y1 = y1;
            this.y2 = y2;
            this.y3 = y3;
            this.y4 = y4;

        }
        public void SetCoordinates(double x1, double y1, double x2, double y2, double x3, double y3, double x4, double y4)
        {
            this.x1 = x1;
            this.x2 = x2;
            this.x3 = x3;
            this.x4 = x4;
            this.y1 = y1;
            this.y2 = y2;
            this.y3 = y3;
            this.y4 = y4;
        }
        //make getters
        public double X1 => x1;
        public double Y1 => y1;
        public double X2 => x2;
        public double Y2 => y2;
        public double X3 => x3;
        public double Y3 => y3;
        public double X4 => x4;
        public double Y4 => y4;
    
    public bool CheckEqualSides()
    {
            double[] sides = new double[4]
            {
                Math.Sqrt(Math.Pow(x2 - x1, 2) + Math.Pow(y2 - y1, 2)),
                Math.Sqrt(Math.Pow(x3 - x2, 2) + Math.Pow(y3 - y2, 2)),
                Math.Sqrt(Math.Pow(x4 - x3, 2) + Math.Pow(y4 - y3, 2)),
                Math.Sqrt(Math.Pow(x1 - x4, 2) + Math.Pow(y1 - y4, 2))
            };

            return sides.All(side => side == sides[0]);
    }
        public double ResPerimeter()
        {
            double sideA = Math.Sqrt(Math.Pow(x2 - x1, 2) + Math.Pow(y2 - y1, 2));
            
            return sideA * 4;
        }
        public double ResArea()
        {
            double sideA = Math.Sqrt(Math.Pow(x2 - x1, 2) + Math.Pow(y2 - y1, 2));
            
            return sideA * sideA;
        }
    }
}

File: C++/Square.cpp
#include "Square.h"
#include <cmath>

Square::Square()
: x1(0), y1(0), x2(0), y2(0), x3(0), y3(0), x4(0), y4(0)
{
}

Square::Square(double x1, double y1, double x2, double y2, double x3, double y3, double x4, double y4)
: x1(x1), y1(y1), x2(x2), y2(y2), x3(x3), y3(y3), x4(x4), y4(y4)
{
}

//setters
void Square::setCoordinates(double X1, double Y1, double X2, double Y2, double X3, double Y3, double X4, double Y4)
{
    x1 = X1;
    y1 = Y1;
    x2 = X2;
    y2 = Y2;
    x3 = X3;
    y3 = Y3;
    x4 = X4;
    y4 = Y4;
}

//getters
double Square::getX1() const {return x1;}
double Square::getY1() const {return y1;}
double Square::getX2() const {return x2;}
double Square::getY2() const {return y2;}
double Square::getX3() const {return x3;}
double Square::getY3() const {return y3;}
double Square::getX4() const {return x4;}
double Square::getY4() const {return y4;}

double Square::resPerimeter() const
{
    double side1 = sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2));
    double side2 = sqrt(pow(x3 - x2, 2) + pow(y3 - y2, 2));
    double side3 = sqrt(pow(x4 - x3, 2) + pow(y4 - y3, 2));
    double side4 = sqrt(pow(x1 - x4, 2) + pow(y1 - y4, 2));
    return side1 + side2 + side3 + side4;
}

double Square::resArea() const
{
    double side1 = sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2));
    double side2 = sqrt(pow(x3 - x2, 2) + pow(y3 - y2, 2));
    return side1 * side2;
}
bool Square::isSquare() const
{
    double side1 = sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2));
    double side2 = sqrt(pow(x3 - x2, 2) + pow(y3 - y2, 2));
    double side3 = sqrt(pow(x4 - x3, 2) + pow(y4 - y3, 2));
    double side4 = sqrt(pow(x1 - x4, 2) + pow(y1 - y4, 2));
    return side1 == side2 && side2 == side3 && side3 == side4;
}